### PR TITLE
=doc Fix Config links in testing (#25421)

### PR DESCRIPTION
**Pull request by jlprat**
refs: #25421
Fixes links to Configuration html page
Removes link to Ray Roestenburg blog page as it's now returning 404
Fixes link to `ConfigFactory` in Configuration doc page

### CODE REVIEW

File: akka-docs/src/main/paradox/testing.md [hunk 1]
```diff
@@ -815,7 +815,7 @@ to the @ref:[reference configuration](general/configuration.md#config-akka-testk
 
 ## Example
 
-Ray Roestenburg's example code from [his blog](http://roestenburg.agilesquad.com/2011/02/unit-testing-akka-actors-with-testkit_12.html) adapted to work with Akka 2.x.
```
Comment (ktoso) [inline akka-docs/src/main/paradox/testing.md]: I've asked @RayRoestenburg about what to do about the link, though I think the drop of the link is perhaps fine... 
Though we could link to the archive post if we wanted to: https://web.archive.org/web/20180114133958/http://roestenburg.agilesquad.com/2011/02/unit-testing-akka-actors-with-testkit_12.html

File: akka-docs/src/main/paradox/testing.md [hunk 2]
```diff
@@ -714,13 +714,13 @@ options:
...
 * *Logging of message invocations on certain actors*
-   This is enabled by a setting in the [Configuration]() — namely
+   This is enabled by a setting in the @ref:[Configuration](general/configuration.md#config-akka-actor) — namely
 `akka.actor.debug.receive` — which enables the `loggable`
 statement to be applied to an actor’s `receive` function:
 
 @@snip [TestkitDocSpec.scala]($code$/scala/docs/testkit/TestkitDocSpec.scala) { #logging-receive }
 
-If the aforementioned setting is not given in the [Configuration](), this method will
+If the aforementioned setting is not given in the @ref:[Configuration](general/configuration.md#config-akka-actor), this method will
```
Comment (ktoso) [inline akka-docs/src/main/paradox/testing.md]: thanks!

**Review by ktoso — APPROVED** [review]
LGTM, pending question to Ray :)

File: akka-docs/src/main/paradox/testing.md (hunk 1 above)
Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: Waiting for the reply then 🙂

Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: I will rephrase it stating that this is an archive link and add the link to it.

Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: Updated

**Review by ktoso — APPROVED** [review]
Looks good, might as well link to the archive I guess ;-)

**Pull request merged**